### PR TITLE
Skip AppleScript actions if Automation Access isn't granted.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Closes https://github.com/Homebrew/homebrew-cask/issues/50413.

### CODE REVIEW

File: Library/Homebrew/utils/user.rb [hunk 1]
```diff
@@ -6,6 +6,21 @@
 require "system_command"
 
 class User < DelegateClass(String)
+  def self.automation_access?
+    *_, status = system_command "osascript", args: [
+      "-e", "with timeout of 0.1 seconds",
```
Comment (vitorgalvao) [inline Library/Homebrew/utils/user.rb]: Has this timeout been reliable in your tests? Reason I ask: https://github.com/Homebrew/homebrew-cask/issues/19076#issuecomment-421836262

**Comment by vitorgalvao** [comment]
Would this be a good PR to also deal with https://github.com/Homebrew/homebrew-cask/issues/19076?

**Comment by reitermarkus** [comment]
Yes, I think this also fixes https://github.com/Homebrew/homebrew-cask/issues/19076.

File: Library/Homebrew/utils/user.rb (hunk 1 above)
Comment (reitermarkus) [reply Library/Homebrew/utils/user.rb]: In my tests, even `0.00001 seconds` was reliable.

Comment (reitermarkus) [reply Library/Homebrew/utils/user.rb]: Increased it to 0.5 since it seems to take a tad longer the first time after a full reboot.

**Pull request merged**